# Практическое задание №3 по теме "Основы математической статистики. Количественные характеристики популяции. Графическое представление данных".

__Задача 1__

Даны значения зарплат из выборки выпускников:

`100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150`

Используя только встроенные питоновские функции и структуры данных (т.е. без библиотек numpy, pandas и др.), посчитайте (несмещённое) среднее квадратичное отклонение для данной выборки.

___Решение___

In [1]:
import math 

$$\overline{X} = \dfrac{1}{n} \displaystyle\sum_{i = 1}^n x_i$$

In [2]:
x = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]
x = sorted(x)
x_len = len(x)
x_mean = sum(x)/x_len
x_mean

73.14285714285714

$$\sigma_X^2 = \dfrac{1}{n - 1} \displaystyle\sum_{i = 1}^n \left( x_i - \overline{X} \right)^2$$

In [3]:
su = [(x_i-x_mean)**2 for x_i in x]
delta_sq = sum(su)/(x_len-1)
delta_sq

2241.828571428571

In [4]:
delta = round(math.sqrt(delta_sq), 8)
delta

47.34795214

___Ответ:___ $\sigma_X =  47.34795214$

---

__Задача 2__

Найдите число выбросов в выборке из задачи 1. Для определения выбросов используйте методику как при построении “усов” в _boxplot_, однако, как и в задаче 1, пользоваться можно только встроенными функциями и структурами данных.

___Решение___

Границами ящика служат первый и третий квартили ($Q_1$, $Q_3$)

Границы усов $[X1, X2]$ расчитываются следующим образом:

$$\left[ Q_1 - 1.5 \times IQR, \: Q_3 + 1.5 \times IQR \right],$$

где $IQR = Q_3 - Q_1$ — интерквартильное расстояние.

Важно при этом, что длину верхнего уса ограничиваем максимальным значением по выборке, попадающим в верхнюю границу уса; длину нижнего уса ограничиваем минимальным значением по выборке, попадающим в длину нижнего уса.

In [5]:
q1 = x[x_len // 4]
q3 = x[3 * x_len // 4]
iqr = q3 - q1

q1, q3

(45, 84)

In [6]:
x1 = q1 - 1.5 * iqr
x2 = q3 + 1.5 * iqr

for x_i in x:
    if x1 < x_i:
        x1 = x_i
        break

        
for x_i in x[::-1]:
    if x2 > x_i:
        x2 = x_i
        break

print(f"Х1={x1}, Х2={x2}")

Х1=17, Х2=100


In [7]:
outlier = [x_i for x_i in x if (x_i > x2 or x_i < x1)]
print(f"Количество выбросов: {len(outlier)}")

Количество выбросов: 2


___Ответ:___ 2 выброса.

---

__Задача 3__

В университет на факультет **A** поступило 100 человек, на факультет **B** - 90 человек и на факультет **C** - 45 человек. Вероятность того, что студент с факультета **A** сдаст первую сессию, равна 0.6. Для студента с факультета **B** эта вероятность равна 0.7, а для студента с факультета **C** - 0.8. Случайным образом выбирается студент, сдавший первую сессию. Какое событие наиболее вероятно:

<ol>
    <li>студент учится на факультете A,</li>
    <li>студент учится на факультете B,</li>
    <li>студент учится на факультете C?</li>
</ol>

__Решение__

Сразу же определимся с тем, какие события мы здесь рассматриваем. Пусть событие $A$ заключается в том, что студент сдал сессию. Кроме того, пусть для каждого $i \in \{ 1, 2, 3 \}$ событие $B_i$ заключается в том, что студент учится на $i$-м факультете.

Распишем чему равна вероятность того, что выбранный студент обучается на $i$-м факультете:

$$P(B_1) = \frac{100}{235}$$, $$P(B_2) = \frac{90}{235}$$, $$P(B_3) = \frac{45}{235}$$

In [8]:
p_b_1 = 100/235
p_b_2 = 90/235
p_b_3 = 45/235

Распишем чему равны вероятности успешной сдачи сессии студентом, если известно, что он обучается на $i$-м факультете:

$$P(A|B_1) = 0.6$$, $$P(A|B_2) = 0.7$$, $$P(A|B_3) = 0.8$$

In [9]:
p_a_b_1 = 0.6
p_a_b_2 = 0.7
p_a_b_3 = 0.8

Искомые вероятности можно найти с помощью формулы Байеса:

$$P(B_1|A) = \frac{P(A|B_1) \cdot P(B_1)}{P(A)}$$, $$P(B_2|A) = \frac{P(A|B_2) \cdot P(B_2)}{P(A)}$$, $$P(B_3|A) = \frac{P(A|B_3) \cdot P(B_3)}{P(A)}$$

Найдем $P(A)$ с помощью формулы полной вероятности:

$$P(A) = P(B_1) \cdot P(A | B_1) + P(B_2) \cdot P(A | B_2) + P(B_3) \cdot P(A | B_3) = $$

$$ 0.6\cdot \frac{100}{235} + 0.7\cdot \frac{90}{235} + 0.8\cdot \frac{45}{235} = 0.67659574$$

In [10]:
p_a = p_b_1*p_a_b_1 + p_b_2*p_a_b_2 + p_b_3*p_a_b_3
round(p_a, 8)

0.67659574

Находим искомые вероятности:

$$P(B_1|A) = \frac{P(A|B_1) \cdot P(B_1)}{P(A)} = \frac{0.6 \cdot 100}{235 \cdot 0.67659574} = 0.37735849$$

$$P(B_2|A) = \frac{P(A|B_2) \cdot P(B_2)}{P(A)} = \frac{0.7 \cdot 90}{235 \cdot 0.67659574} = 0.39622642$$

$$P(B_3|A) = \frac{P(A|B_3) \cdot P(B_3)}{P(A)} = \frac{0.8 \cdot 45}{235 \cdot 0.67659574} = 0.22641509$$

In [11]:
p_b_1_a = round(p_a_b_1*p_b_1/p_a, 8)
p_b_2_a = round(p_a_b_2*p_b_2/p_a, 8)
p_b_3_a = round(p_a_b_3*p_b_3/p_a, 8)
p_b_1_a, p_b_2_a, p_b_3_a

(0.37735849, 0.39622642, 0.22641509)

___Ответ:___ 2. Наиболее вероятно, что студент учится на факультете B.

---